src/database-layer/db.ts

```diff
@@ -196,14 +216,21 @@ export abstract class Tx {
 
 		const closeTransaction = (message: string) => {
 			pendingExecutes.cancel()
-			const rejectionValue = new Error(message)
-			// We return a new rejected promise on each call so that bluebird can handle
-			// logging errors if the rejection is not handled (but only if it is not handled)
-			this.executeSql = (_sql, _bindings, callback) => {
-				return Promise.reject(rejectionValue).nodeify(callback)
-			}
-			this.rollback = this.end = (callback) => {
-				return Promise.reject(rejectionValue).nodeify(callback)
+			if (DEBUG) {
+				// In debug mode we create the error here to give useful stack traces,
+				// but it adds significant overhead for a production environment
+				const rejectionValue = new Error(message)
+				// We return a new rejected promise on each call so that bluebird can handle
+				// logging errors if the rejection is not handled (but only if it is not handled)
+				this.executeSql = (_sql, _bindings, callback) => {
+					return Promise.reject(rejectionValue).nodeify(callback)
+				}
+				this.rollback = this.end = (callback) => {
+					return Promise.reject(rejectionValue).nodeify(callback)
+				}
+			} else {
+				this.executeSql = rejectedExecuteSql(message)
+				this.rollback = this.end = rejectedRollbackEnd(message)
```

Comment:
Would it make sense to combine the two methods into one (returning a composite object), that could possibly reuse the created error (similarly to what we do in `if (DEBUG)`)?

Reply:
Done